datafusion/core/src/datasource/file_format/mod.rs

```diff
@@ -63,8 +67,9 @@ pub trait FileFormat: Send + Sync + fmt::Debug {
     /// TODO: should the file source return statistics for only columns referred to in the table schema?
     async fn infer_stats(
         &self,
-        reader: Arc<dyn ObjectReader>,
+        store: &dyn ObjectStore,
```

Comment:
what is the reason for the change to `&dyn ObjectStore`? Couldn't this make it harder to pass the object store on to an inner implementation (that might need an `Arc<ObjectStore>`)?

Reply:
```suggestion
        store: &Arc<dyn ObjectStore>,
```

Reply:
Predominantly to avoid being one of the implementations "that might need an `Arc<ObjectStore>`". I didn't see a compelling reason why this method would need a "ownable" reference, but given the rest of the codebase is pretty hard-coded to `Arc<dyn ObjectStore>` I don't feel strongly. Will change back :smile: